hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+                        if res_obj.metadata.originalcoverage.all().first():
+                            res_obj.metadata.originalcoverage.all().delete()
+                        if res_obj.metadata.coverages.all().first():
+                            res_obj.metadata.coverages.all().delete()
+
+                        tmp_dir=rslt['tmp_dir']
+                        baseFilename=rslt['baseFilename']
+                        uploaded_file_type=rslt['uploaded_file_type']
+                        uploadedFileCount=rslt['uploadedFileCount']
+                        uploadedFilenameString=rslt['uploadedFilenameString']
+                        shp_full_path=rslt['shp_full_path']
+                        parsed_md_dict=None
+                        if uploaded_file_type == "shp" or uploaded_file_type == "zipped_shp":
+                            meta_array, meta_dict=parse_shp_zshp(uploaded_file_type, baseFilename, uploadedFileCount, uploadedFilenameString, shp_full_path)
+
+                        #create metadat objects
+                        if meta_dict.has_key("coverage"):
+                            res_obj.metadata.create_element('Coverage', type=meta_dict["coverage"]['Coverage']['type'], value=meta_dict["coverage"]['Coverage']['value'])
+                        res_obj.metadata.create_element('OriginalFileInfo',
+                                                        fileType=meta_dict["originalfileinfo"]['fileType'],
+                                                        baseFilename=meta_dict["originalfileinfo"]['baseFilename'],
+                                                        fileCount=meta_dict["originalfileinfo"]['fileCount'],
+                                                        filenameString=meta_dict["originalfileinfo"]['filenameString'])
+                        res_obj.metadata.create_element('OriginalCoverage',
+                                                        northlimit=meta_dict["originalcoverage"]['originalcoverage']['northlimit'],
+                                                        southlimit =meta_dict["originalcoverage"]['originalcoverage']['southlimit'],
+                                                        westlimit=meta_dict["originalcoverage"]['originalcoverage']['westlimit'],
+                                                        eastlimit=meta_dict["originalcoverage"]['originalcoverage']['eastlimit'],
+                                                        projection_string=meta_dict["originalcoverage"]['originalcoverage']['projection_string'],
+                                                        projection_name=meta_dict["originalcoverage"]['originalcoverage']['projection_name'],
+                                                        datum=meta_dict["originalcoverage"]['originalcoverage']['datum'],
+                                                        unit =meta_dict["originalcoverage"]['originalcoverage']['unit'])
```

Comment:
spacing issues again.
Also you could probably simplify maybe of these deep dictionary lookups by storing, for example, `meta_dict["originalcoverage"]["originalcoverage"]` in a local variable